include/dd/ComputeTable.hpp

```diff
@@ -61,7 +60,9 @@ namespace dd {
             if (entry.rightOperand != rightOperand) return result;
 
             if constexpr (std::is_same_v<RightOperandType, dEdge>) {
-                if ((entry.result.p->flags < 7) == useDensityMatrix) return result;
+                // Since density matrices are reduced representations of matrices, a density matrix may not be returned when a matrix is required. The statement isDensityMatrix(entry.result.p.flags())
+                // evaluates true if the result is a density matrix. In this case the result is only returned when a density matrix is required (=!useDensityMatrix)
+                if ((dd::dEdge::isDensityMatrixNode((std::int_fast8_t) entry.result.p->flags)) && !useDensityMatrix) return result;
```

Comment:
This should rather be something like
```suggestion
                if (entry.result.p->isDensityMatrixNode() && !useDensityMatrix) return result;
```

Reply:
I simplified accessing the flag variable using dedicated functions.